### ISSUE TITLE
$state.go('/list', {aParam: null}) throws TypeError: Cannot read property 'toString' of null.

### ISSUE THREAD

**Issue by Vohidjon** [angular-ui/ui-router#3197]
angular -v : 1.5.8
ui-router -v: 0.3.2

$stateProvider.state({
  url: '/list?{aParam:int}',
  ......
});
$state.go('/list', {aParam: 13}) works
$state.go('/list', {aParam: undefined}) works
$state.go('/list', {aParam: null}) throws error

Is this expected behavior? If it is, could you direct me to docs or guides. If it is not, i would like to provide a plunker.

**Comment by christopherthielen** [comment]
If anything, I would expect that to throw an error about the param value being invalid.

That's a bug, thanks for reporting it!

Here's the problem: 

```js
    "int": {
      encode: valToString,
      decode: function(val) { return parseInt(val, 10); },
      is: function(val) { return isDefined(val) && this.decode(val.toString()) === val; },
      pattern: /\d+/
    },
    ```
We should check if val `!== null` in the `is()` function

**Comment by christopherthielen** [comment]
This is also a bug in 1.0, so tagging appropriately. After the fix goes into 1.0 it should be backported to legacy for the 0.4.0 release.

**Comment by christopherthielen** [comment]
Closed by aa551e4252e784a49fda93865f1d1c03b4e36017 in 1.0

**Comment by christopherthielen** [comment]
Fixed by 	61728d7 in 0.4.0
